mesonbuild/environment.py

```diff
@@ -409,6 +413,8 @@ def get_gnu_compiler_type(defines):
         # Detect GCC type (Apple, MinGW, Cygwin, Unix)
         if '__APPLE__' in defines:
             return GCC_OSX
+        elif '_AIX' in defines: # Like OSX, AIX uses traditional Unix (non-gnu) ld
+            return GCC_OSX
```

Comment:
This seems a bit suspect. Does AIX's linker really behave in the same way as OSX's?